### ISSUE TITLE
Method "Shopware::PasswordEncoder" not found failure in Shopware/Components\Auth\Adapter\Default.php on line 218

### ISSUE THREAD

**Issue by dsp84** [shopware5/shopware#88]
https://github.com/ShopwareAG/shopware-4/blob/master/engine/Shopware/Components/Auth/Adapter/Default.php#L218

<h3>Method "Shopware::PasswordEncoder" not found failure in Shopware/Components\Auth\Adapter\Default.php on line 218</h3>
    

```
<h3>Stack trace:</h3>
<div style="overflow:auto;">
<pre>#0 Shopware/Components\Auth\Adapter\Default.php(218): Enlight_Application-&gt;__call(&#039;PasswordEncoder&#039;, Array)
```
#1 Shopware/Components\Auth\Adapter\Default.php(218): Shopware-&gt;PasswordEncoder()
#2 Zend\Auth\Adapter\DbTable.php(386): Shopware_Components_Auth_Adapter_Default-&gt;_authenticateValidateResult(Array)
#3 Enlight\Components\Auth\Adapter\DbTable.php(142): Zend_Auth_Adapter_DbTable-&gt;authenticate()
#4 Shopware/Components\Auth\Adapter\Default.php(109): Enlight_Components_Auth_Adapter_DbTable-&gt;authenticate()
#5 Zend\Auth.php(117): Shopware_Components_Auth_Adapter_Default-&gt;authenticate()
#6 Enlight\Components\Auth.php(74): Zend_Auth-&gt;authenticate(Object(Shopware_Components_Auth_Adapter_Default))
#7 Shopware/Components\Auth.php(136): Enlight_Components_Auth-&gt;authenticate(Object(Shopware_Components_Auth_Adapter_Default))
#8 Shopware/Components\Auth.php(92): Shopware_Components_Auth-&gt;authenticate(Object(Shopware_Components_Auth_Adapter_Default))
#9 Shopware/Controllers\Backend\Login.php(64): Shopware_Components_Auth-&gt;login(&#039;admin&#039;, &#039;dspdevel&#039;)
#10 Enlight\Controller\Action.php(148): Shopware_Controllers_Backend_Login-&gt;loginAction()
#11 Enlight\Controller\Dispatcher\Default.php(521): Enlight_Controller_Action-&gt;dispatch(&#039;loginAction&#039;)
#12 Enlight\Controller\Front.php(214): Enlight_Controller_Dispatcher_Default-&gt;dispatch(Object(Enlight_Controller_Request_RequestHttp), Object(Enlight_Controller_Response_ResponseHttp))
#13 Shopware/Components\HttpCache\HttpKernel.php(79): Enlight_Controller_Front-&gt;dispatch()
#14 Symfony\Component\HttpKernel\HttpCache\HttpCache.php(434): Shopware\Components\HttpCache\HttpKernel-&gt;handle(Object(Symfony\Component\HttpFoundation\Request), 1, true)
#15 Shopware/Components\HttpCache\AppCache.php(234): Symfony\Component\HttpKernel\HttpCache\HttpCache-&gt;forward(Object(Symfony\Component\HttpFoundation\Request), true, NULL)
#16 Symfony\Component\HttpKernel\HttpCache\HttpCache.php(230): Shopware\Components\HttpCache\AppCache-&gt;forward(Object(Symfony\Component\HttpFoundation\Request), true)
#17 Shopware/Components\HttpCache\AppCache.php(99): Symfony\Component\HttpKernel\HttpCache\HttpCache-&gt;pass(Object(Symfony\Component\HttpFoundation\Request), true)
#18 Shopware/Bootstrap.php(70): Shopware\Components\HttpCache\AppCache-&gt;handle(Object(Symfony\Component\HttpFoundation\Request))
#19 Enlight\Application.php(192): Shopware_Bootstrap-&gt;run()
#20 shopware.php(83): Enlight_Application-&gt;run()
#21 {main}</pre>

```
</div>
```

<div class="doublespace">&nbsp;</div>

**Comment by dnoegel** [comment]
Hi @dsp84,

please provide some context, just posting the exception without a word is not that helpful :).

You might want to apply the deltas, though.

**Comment by dsp84** [comment]
Installed a fresh Shopware 4.1, fresh checkout. 
Error occurs during Backend Login.

No additional Context needed. If u resolve the link to the file in your master you will soon find the error.

**Comment by bcremer** [comment]
Hi @dsp84,

I just installed a fresh master, no problems here.
Make sure you used `ant build-database` so database-deltas are applied.

**Comment by dsp84** [comment]
Problem solved.
Cross-Plattform problem win<>*nix.

We created a step between ur native build step to fit the OS reqs, that did it.
Indeed, windows is not the usual way for devs but actually we can't replace that machine, maybe after release during the refactoring :-)

---

Example:
Symlinks on Win will not resolved with ant (similar problems with other tools), ByteCode will still concatinated and there are a unresolvable condition on win7x64/mysql5.6, but the build will still success.

In fact, the generated sql file is broken, still executed with no error, build finished and the Shop is flying high ... We recognized several php fatal errors, but did no analysation.

All other OS works properly.
